lib/index.js

```diff
@@ -276,3 +266,48 @@ module.exports = {
     }
   }
 }
+
+const baseConfig = { plugins: { vue: mod } }
+
+mod.configs = {
+  // eslintrc configs
+  base: require('./configs/base'),
+  essential: require('./configs/vue2-essential'),
+  'no-layout-rules': require('./configs/no-layout-rules'),
+  recommended: require('./configs/vue2-recommended'),
+  'strongly-recommended': require('./configs/vue2-strongly-recommended'),
+  'vue3-essential': require('./configs/vue3-essential'),
+  'vue3-recommended': require('./configs/vue3-recommended'),
+  'vue3-strongly-recommended': require('./configs/vue3-strongly-recommended'),
+
+  // flat configs
+  'flat/base': Object.assign(baseConfig, require('./configs/flat/base.js')),
```

Comment:
Wouldn't this result in them sharing the same object? It might be safe to use spread instead.

```suggestion
  'flat/base': {...baseConfig, ...require('./configs/flat/base.js')},
```

Reply:
good catch! 👍  
should be addressed in 95ddc6f4c65555c3e8fa0b027b22f0d4fbbc04fb.